neonvm/controllers/virtualmachine_controller.go

```diff
@@ -321,6 +321,11 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 	// Generate runner pod name
 	if len(virtualmachine.Status.PodName) == 0 {
 		virtualmachine.Status.PodName = names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-", virtualmachine.Name))
+		// Update the .Status.PodName on API Server to avoid creating multiple pods for a single VM
```

Comment:
nit: it can also update `.Status.SSHSecretName`. So technically we update not the particular field, but the entire `virtualmachine` object. Maybe the wording here and in the error message can be improved.